plugins/upload/src/lib.rs

```diff
@@ -72,44 +72,50 @@ async fn download(
     body: Option<String>,
     on_progress: Channel<ProgressPayload>,
 ) -> Result<()> {
-    let client = reqwest::Client::new();
-    let mut request = if let Some(body) = body {
-        client.post(url).body(body)
-    } else {
-        client.get(url)
-    };
-    // Loop trought the headers keys and values
-    // and add them to the request object.
-    for (key, value) in headers {
-        request = request.header(&key, value);
-    }
+    let url = url.to_string();
+    let file_path = file_path.to_string();
```

Comment:
Might as well change the function arguments then. PathBuf implement Deserialize as well so if we need that later we could use that instead of a string.

Reply:
Good idea. Done.